provider.go

```diff
@@ -1052,55 +1072,357 @@ func (p *provider) Delete(ctx context.Context, req *rpc.DeleteRequest) (*emptypb
 
 }
 
+// ConstructRequest captures enough data to be able to register nested components against the caller's resource
+// monitor.
 type ConstructRequest struct {
-	URN       presource.URN
-	Preview   bool
-	Construct func(context.Context, ConstructFunc) (ConstructResponse, error)
```

Comment:
Note that the field names are coming into alignment with resource.proto, and that `URN` changes to `Urn` to match the other request types in this package.